lib/modules/manager/poetry/extract.spec.ts

```diff
@@ -169,19 +170,96 @@ describe('modules/manager/poetry/extract', () => {
       });
     });
 
+    it('parses git dependencies long commit hashs on http urls', async () => {
+      const content = codeBlock`
+        [tool.poetry.dependencies]
+        fastapi = {git = "https://github.com/tiangolo/fastapi.git", rev="6f5aa81c076d22e38afbe7d602db6730e28bc3cc"}
+        werkzeug = ">=0.14"
+      `;
+      const res = (await extractPackageFile(content, filename))!.deps;
+      expect(res).toHaveLength(2);
+      expect(res[0]).toMatchObject({
+        depName: 'fastapi',
+        datasource: GitRefsDatasource.id,
+        currentValue: undefined,
+        currentDigest: '6f5aa81c076d22e38afbe7d602db6730e28bc3cc',
+        replaceString: '6f5aa81c076d22e38afbe7d602db6730e28bc3cc',
+        pinDigests: true,
+        packageName: 'https://github.com/tiangolo/fastapi.git',
+      });
```

Comment:
See [this test](https://github.com/renovatebot/renovate/pull/26367/files#diff-2c72b66f90a1c4836a08fc027791c4d9d4b5b6cf5601f47e4e5b5295c0c0ca04R136) how to implement this